README.md

```diff
@@ -1 +1,47 @@
-# java-web-servlet-filter
\ No newline at end of file
+# OpenTracing Java Web Servlet Filter Instrumentation
+
+[![Travis](https://travis-ci.org/opentracing-contrib/java-web-servlet-filter.svg?branch=master)](https://travis-ci.org/opentracing-contrib/java-web-servlet-filter)
+
+This library provides instrumentation for Java Web Servlet applications.
+
+## Maven
+```xml
+   <dependency>
+     <groupId>io.opentracing.contrib</groupId>
+     <artifactId>opentracing-web-servlet-filter</artifactId>
+     <version>0.0.1.SNAPSHOT</version>
+   </dependency>
+```
+
+## Initialization
+
+Tracing filter can be programmatically initialized:
+```java
+   TracingFilter filter = new TracingFilter(tracer, Arrays.asList(SpanDecorator.STANDARD_TAGS, TracingDecision.TRACE_ALL))
+   servletContext.addFilter("tracingFilter", filter);
+```
+
+or added in `web.xml` but it requires to add several properties to ServletContext attributes.
+```java
+   servletContext.setAttribute(TracingFilter.TRACER, tracer);
+   servletContext.setAttribute(TracingFilter.SPAN_DECORATORS, decorators); // optional, if no present SpanDecorator
+   .STANDARD_TAGS is applied
+   servletContext.setAttribute(TracingFilter.TRACING_DECISION, tracingDecision); // optional
+```
+
+## Accessing Server Span
+Current server span is accessible in HttpServletRequest attributes.
+```java
+   Span span = (Span)httpservletRequest.getAttribute(TracingFilter.ACTIVE_SPAN);
```

Comment:
I guess you could create a static method on `TracingFilter` that takes a servlet request and encapsulates the cast... YMMV, no strong feelings.